great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -289,8 +290,33 @@ def get_batch_data_and_markers(
             reader_method: str = batch_spec.reader_method
             reader_options: dict = batch_spec.reader_options or {}
             path: str = batch_spec.path
+            schema: Optional[
+                Union[pyspark.sql.types.StructType, dict, str]
+            ] = reader_options.get("schema")
+
+            # schema can be a dict if it has been through serialization step,
+            # either as part of the datasource configuration, or checkpoint config
+            if isinstance(schema, dict):
+                schema: pyspark.sql.types.StructType = sparktypes.StructType.fromJson(
+                    schema
+                )
+
+            # this can happen if we have not converted schema into json at Datasource-config level
+            elif isinstance(schema, str):
+                raise ge_exceptions.ExecutionEngineError(
+                    f"""
+                                Spark schema was not properly serialized.
+                                Please run the .jsonValue() method on the schema object before loading into GE.
+                                schema: your_schema.jsonValue()
+                                """
+                )
             try:
-                reader: DataFrameReader = self.spark.read.options(**reader_options)
+                if schema:
```

Comment:
Is `self.spark.read.options(**reader_options)` equivalent to `self.spark.read.schema(None).options(**reader_options)`? I'm just wondering if we can get rid of the conditional.

Reply:
I was looking at [the source via the docs](https://spark.apache.org/docs/3.1.1/api/python/_modules/pyspark/sql/readwriter.html#DataFrameReader.schema) and it looks like it can't be None.

Reply:
Another great suggestion. Looking into this now

Reply:
`TypeError: schema should be StructType or string`

Reply:
Yup, this is what I see too